### DISCUSSION TITLE
Timeline block with animated line (Interactivity API)

### DISCUSSION THREAD

**Discussion by colorful-tones** [WordPress/developer-blog-content#240] [Topic Ideas]
I feel inspired by Keith Devon's (co-founder of [Highrise Digital](https://highrise.digital/)) original Twitter (X) share here: https://x.com/keithdevon/status/1764938947233263861?s=20 

He shared a neat timeline feature he built. It is a series of grouped elements stacked vertically in reverse chronological order. Each grouped timeline event has a date, a brief description, and a picture. As the user scrolls down the page, a vertical line visually ties each event on the timeline together by animating down the page (when in scroll view).

This could be tied to a CSS class and applied to a Group or Query block wrapper, in which we could use a block variation to extend core blocks to apply the class and then utilize the Interactivity API to apply the animated line.

This would tie together a lot of new and recent stuff coming out, and I would be happy to dive in.

**Comment by colorful-tones** [comment]
Here are some screen grabs from Keith's original post. I'm awaiting to see whether he approves of attaching these visuals for inspiration only. [See X (Twitter)](https://x.com/dcook/status/1768713975137501205?s=20)

Keep in mind, I do not propose that we use these visuals in the final tutorial, but mostly as a starting inspirational wireframe to drive the final implementation.

![Screenshot 2024-03-15 at 2 48 56 PM](https://github.com/WordPress/developer-blog-content/assets/405912/b90c7bac-51ac-4031-b1eb-e84df0dfddbd)
![Screenshot 2024-03-15 at 2 56 44 PM](https://github.com/WordPress/developer-blog-content/assets/405912/d85b0968-674c-494c-9171-daa0a47ffe17)

**Reply by colorful-tones** [reply]
Keith Devin approves 👏 https://x.com/keithdevon/status/1768739727090290931?s=20

**Comment by flexseth** [comment]
Really like the idea! I've seen some really impressive horizontal timelines, and would like to see a horizontally scrolling timeline that can be scrolled similarly to a slider, with touch capabilities. 

Here's a JavaScript library for doing so horizontally
https://timeline.knightlab.com/

<img width="2288" alt="image" src="https://github.com/WordPress/developer-blog-content/assets/3792502/77fd1285-fde1-4d3f-ac81-a4d7c1526c1e">

**Reply by colorful-tones** [reply]
That is a different direction than what was initially proposed. It would certainly be neat to see somebody create and have a demo for this. Still, there would be a lot of accessibility considerations and challenges to work through with that type of implementation, which would increase the scope. Whereas, I believe the original timeline I proposed could likely use all native WordPress blocks and tie in a sprinkle of the Interactivity API and maybe even a few newer APIs to demonstrate to developers how these new features can be pulled together to create elegant, new experiences. 🤔

**Reply by flexseth** [reply]
No worries, Birgit wanted me to comment it here so I did :)

May be right about the accessibility, I didn't even think about that. It would be interesting to build it out and run the markup through an accessibility checker.

**Comment by colorful-tones** [comment]
Just saw another implementation of this from Multidots: https://www.multidots.com/wordpress-contribution/
Bonus: it is using core blocks and just needs to animate that line down the page. 🕺

**Comment by bph** [comment]
This topic was approved, and the conversation continues on the issue 

- #246

**Comment by colorful-tones** [comment]
This has been published. Thanks everyone for your support! 🫶 
https://developer.wordpress.org/news/2024/06/13/how-to-create-an-animated-timeline-plugin/
